run/index.js

```diff
@@ -114,6 +115,15 @@ module.exports = async function run ({ ipc, args, cmdArgs, link, storage, detach
     state.update({ config: await ipc.config() })
 
     const pear = new API(ipc, state)
+
+    pear.messages({ type: 'pear/restart' }, async () => {
+      // Wait for sidecar to shutdown
+      // TODO: Figure out how to properly detect sidecar shutdown from the client
+      await new Promise(resolve => setTimeout(resolve, 1500))
```

Comment:
https://github.com/holepunchto/pear-ipc/blob/main/api.js - add restart to api.js and use the lock file approach - then just called Pear.restart